pyqir/pyqir/_native.pyi

```diff
@@ -1223,13 +1243,17 @@ def if_result(
     ...
 
 def add_string_attribute(
-    function: Function, kind: str, value: Optional[str] = None
+    function: Function,
+    kind: str,
+    value: Optional[str] = None,
+    index: Optional[int] = None,
 ) -> bool:
     """
     Adds a string attribute to the given function.
 
     :param function: The function.
     :param key: The attribute key.
     :param value: The attribute value.
+    :param index: The optional attribute index, defaults to the function index.
```

Comment:
Since there are just two indexes, mentioning them here and briefly describing them would be helpful.

Reply:
I misunderstood the possible values this param could take. Closing since it's no longer relevant.